functionaltests/src/test/java/specs/paymentmethods/sofortueberweisung/ChargeImmediatelyFixture.java

```diff
@@ -174,24 +165,42 @@ public String createPayment(final String paymentName,
     public boolean executeRedirectForPayments(final String paymentNames) throws ExecutionException {
         final Collection<String> paymentNamesList = ImmutableList.copyOf(thePaymentNamesSplitter.split(paymentNames));
 
-        paymentNamesList.forEach(paymentName -> {
-            final Payment payment = fetchPaymentByLegibleName(paymentName);
-            try {
-                Optional.ofNullable(payment.getCustom())
-                        .map(customFields -> customFields.getFieldAsString(CustomFieldKeys.REDIRECT_URL_FIELD))
-                        .map(redirectCustomField -> webDriver.executeSofortueberweisungRedirect(redirectCustomField,
-                                                                                getTestDataSwBankTransferIban(),
-                                                                                getTestDataSwBankTransferPin(),
-                                                                                getTestDataSwBankTransferTan())
-                                .replace(baseRedirectUrl, "[...]"))
-                        .ifPresent(successUrl -> successUrlForPayment.put(paymentName, successUrl));
-            } catch (Exception e) {
-                LOG.error("Error executing redirect for Sofortüberweisung Charge Immediate fixture", e);
-            }
-
-        });
-
-        return successUrlForPayment.size() == paymentNamesList.size();
+        // run all 3 payments approval in parallel, aka 3 different sessions
+        final int processedCount = paymentNamesList.stream().parallel()
+                .mapToInt(this::approvePaymentAsCustomer)
+                .sum();
+
+        return successUrlForPayment.size() == processedCount;
+    }
+
+    /**
+     * Simulate customer's payment approval: log in in browser, set field values, press submit buttons.
+     *
+     * @param paymentName payment name to approve
+     * @return 1 if payment is approved, 0 - if approval was not possible.
+     */
+    private int approvePaymentAsCustomer(String paymentName) {
+        final Payment payment = fetchPaymentByLegibleName(paymentName);
```

Comment:
What kind of payment can be here? Isn't this method testing only Sofort payment method?

Reply:
different amount/currency/bank account settings